apps/web/src/pages/Applications/ApplicationLayout.tsx

```diff
@@ -31,6 +31,7 @@ import { ContextType } from "./useApplication";
 import Application_PoolCandidateFragment from "./fragment";
 import { getApplicationSteps } from "./utils";
 
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
 type RouteParams = {
   experienceId: string;
```

Comment:
The bit in question here is
`useParams<>()`

Will that be touched in https://github.com/GCTC-NTGC/gc-digital-talent/issues/11383 ? That issue only mentions 
`useRequiredParams<>()` 

![image](https://github.com/user-attachments/assets/3000507c-4fc8-4224-bc42-3d727993f86d)

![image](https://github.com/user-attachments/assets/d55e38e0-3410-46c9-8bc3-65315fa7b5c2)

Reply:
uh-oh, no it does not because we actually don't control `useParams` :grimacing: 

Hmmm, I wonder if we need to make a type that satisfies the constraint :thinking:

Reply:
Digging around, we should probably use `useRequiredParams<>()` instead?

Reply:
That one is tough because if you don't provide the param it will error out. For the case of experience ID we actually sometimes don't want ti (create page)

Reply:
Well, I'll leave it to being your call